### PR TITLE
Update switch/tclk/xnet/session examples to display error code

### PR DESCRIPTION
**Pull request by dmondrik**
### What does this Pull Request accomplish?

We have changed how the error code gets back to the user. The examples should show the best practice of handling it.

In particular, as of [this PR which just went in](https://github.com/ni/grpc-device/pull/715), the status code will be part of the trailing metadata.

### Why should this Pull Request be merged?

[Task 2124241](https://ni.visualstudio.com/DevCentral/_workitems/edit/2124241): Update examples (niswitch, nitclk, nixnet, session) to handle new errors

### What testing has been done?

Built locally.

### CODE REVIEW

**Review by danestull — APPROVED** [review]

File: examples/nidcpower/measure-record.py [hunk 1]
```diff
@@ -204,18 +203,15 @@ def _on_close(event):
 
 except grpc.RpcError as rpc_error:
     error_message = rpc_error.details()
+    for key, value in rpc_error.trailing_metadata() or []:
+        if key == "ni-error":
+            error_message += f"\nError status: {value}"
```
Comment (danestull) [inline examples/nidcpower/measure-record.py:208]: Does this value include the error description or message or are we just printing out the code now?

File: examples/nidcpower/measure-record.py [hunk 2]
```diff
@@ -204,18 +203,15 @@ def _on_close(event):
 
 except grpc.RpcError as rpc_error:
     error_message = rpc_error.details()
+    for metadatum in rpc_error.trailing_metadata() or []:
+        if metadatum.key == "ni-error":
+            error_message += f"\nError status: {metadatum.value.decode('utf-8')}"
```
Comment (bkeryan) [inline examples/nidcpower/measure-record.py:208]: Did you retest the examples after adding `decode`? 

Based on the `grpcio` Cython source, I would expect `value` to be a `str` unless `key` ends in "-bin".
https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/_cython/_cygrpc/metadata.pyx.pxi#L61

**Review by reckenro — APPROVED** [review]

File: examples/nidcpower/measure-record.py (hunk 2 above)
Comment (dmondrik) [reply examples/nidcpower/measure-record.py]: OK the code I came up with in the end behaves correctly, passes mypy, and is formatted in a sane readable fashion:
```
    for entry in rpc_error.trailing_metadata() or []:
        if entry.key == "ni-error":
            value = entry.value if isinstance(entry.value, str) else entry.value.decode("utf-8")
            error_message += f"\nError status: {value}"
```
FYI @DavidCurtiss @reckenro @danestull @tdunkle

File: examples/nitclk/multi-device-configured-acquisition-tclk.py [hunk 3]
```diff
@@ -80,23 +80,27 @@
...
+        )
+        sys.stderr.write(f"{warning_message.error_string}\nWarning status: {response.status}\n")
 
 
 try:
-    # Initialize the session
-    i = 1
-    for resource in RESOURCES:
+    # Initialize the sessions
+    for i, resource in enumerate(RESOURCES, start=1):
```
Comment (astarche) [inline examples/nitclk/multi-device-configured-acquisition-tclk.py:103]: This is as good a place as any to comment: these examples look a lot better. Thanks!

**Review by astarche — APPROVED** [review]

File: examples/nidcpower/measure-record.py (hunk 2 above)
Comment (bkeryan) [reply examples/nidcpower/measure-record.py]: I think `isinstance` is the key: it tells mypy that this code can handle `str`.

Apparently `assert isinstance(entry.value, str)` or `cast(str, entry.value)` will also work:
https://mypy.readthedocs.io/en/stable/type_narrowing.html

**Pull request merged**